src/ruby/puppetserver-lib/puppet/server/compiler.rb

```diff
@@ -14,11 +14,40 @@ def compile(request_data)
         node = create_node(processed_hash)
 
         catalog = Puppet::Parser::Compiler.compile(node, processed_hash['job_id'])
+
+        maybe_save(processed_hash, node.facts, catalog)
+
         catalog.to_data_hash
       end
 
       private
 
+      def maybe_save(processed_hash, facts, catalog)
```

Comment:
Could have a more meaningful name than processed_hash here, which doesn't really make sense in this context since it doesn't need to be distinguished in this function from the unprocessed one

(a better design there imo would be to have the java to ruby stuff contained entirely in the file that implements the interface, since that's where the other java stuff is, then everything in this file gets passed the already "processed" data and can just call it something semantically meaningful)

Reply:
Agreed about moving it back out to the interface file.

Reply:
Should I add that on to this PR? I was thinking about rebasing this onto @Magisus logging PR? Can you look at @Magisus ' PR @pcarlisle 	and merge if it seems good to you?

Reply:
I did do this refactor in my PR, so maybe that should be merged first.